### ISSUE TITLE
[Java] Syntax highlighting not working on multi-line method declaration

### ISSUE THREAD

**Issue by alexr00** [atom/language-java#186]
_From @stanleynguyen on November 14, 2018 15:4_

Syntax highlighting doesn't detect language component correctly if a method declaration spans over multiple lines. Here is an example (please only take note of the colors)

Expected:
![capture1](https://user-images.githubusercontent.com/12974927/48491000-925f7000-e861-11e8-9682-3e6142dfdbdf.PNG)

Actual:
![capture](https://user-images.githubusercontent.com/12974927/48491010-968b8d80-e861-11e8-8ab8-5b8dfb92ccf3.PNG)

- VSCode Version: 1.29
- OS Version: Both Windows 10 and MacOS Mojave

Steps to Reproduce:

1. Declare a method with signature that's longer than line width limit
2. Save and wait for the formatter to reformat it to multiple lines

Does this issue occur when all extensions are disabled?: Yes

_Copied from original issue: Microsoft/vscode#63137_

**Comment by alexr00** [comment]
_From @H-C-H on November 28, 2018 19:27_

I can reproduce this error with the following code:
```
class C {
    public
    void foo() {}
}
```
<img width="156" alt="screen shot 2018-11-28 at 20 26 16" src="https://user-images.githubusercontent.com/31404215/49176784-e8e1a980-f34b-11e8-86a7-57b92e05c8e5.png">

`void` and `foo` are not highlighted. The TextMate scope for the declaration is
- meta.class.body.java
- meta.class.java
- source.java

Everything is highlighted correctly when the access modifier `public` is removed. I think the bug is related to the `member-variables` pattern in the syntax definition ([link](https://github.com/Microsoft/vscode/blob/63710859e6ecb2d10c5b6434be73f7e3e1cdff73/extensions/java/syntaxes/java.tmLanguage.json#L848)) which is included in `class-body`.

The C# grammar handles this without problems:
<img width="160" alt="screen shot 2018-11-28 at 20 26 06" src="https://user-images.githubusercontent.com/31404215/49176771-e41cf580-f34b-11e8-96d8-3f37b2df5084.png">

**Comment by felixhao28** [comment]
I think this is due to limitation of the Regex engine oniguruma, which does not work across lines.

**Comment by sadikovi** [comment]
Yes, you are right. But this could be worked around by changing the code for the scope - I will have a look.

**Comment by JoeyRxy** [comment]
so anyone know how to fix it?

**Comment by Eskibear** [comment]
Any updates? BTW hightlighting for multi-line class declaration is also broken:

```java
class 
Foo {
}
```
